sdk/tables/azure-data-tables/azure/data/tables/_entity.py

```diff
@@ -75,17 +76,40 @@ class EntityProperty(object):
     """
 
     def __init__(self,
+                value=None, # type: any
                  type=None,  # type: Union[str,EdmType] # pylint:disable=W0622
-                 value=None  # type: Any
-                 ):
+                #  value=None  # type: Any
+    ):
         """
         Represents an Azure Table. Returned by list_tables.
 
         :param Union[str, EdmType] type: The type of the property.
         :param Any value: The value of the property.
         """
-        self.type = type
+        # self.type = type
         self.value = value
+        if type is not None:
+            self.type = type
+        elif isinstance(value, bytes):
+            self.type = EdmType.BINARY
+        elif isinstance(value, int):
+            self.type = EdmType.INT64
+        elif isinstance(value, uuid.UUID):
+            self.type = EdmType.GUID
+        elif isinstance(value, datetime):
+            self.type = EdmType.DATETIME
+        elif isinstance(value, str):
```

Comment:
we need to make sure this will work for python 2. I think we might need to use six for both the string and binary type

Look here:
https://six.readthedocs.io/